### DISCUSSION TITLE
Multiplayer Prototype

### DISCUSSION THREAD

**Discussion by Gamespleasure** [mbrlabs/Lorien#51] [Show and tell]
Hello! 👋 I just want to show a multiplayer prototype that I've implemented using Godot's ENET.
![Gif](https://user-images.githubusercontent.com/61508621/128642471-4fc696a1-47b1-4ea0-9189-719010d8ddd5.gif)

At the moment it's very barebones, it doesn't shows other users' cursors and the popup is very basic, I mainly made that just to test how hard it would be to implement multiplayer, wasn't that hard! I don't plan on expanding it, It would probably be best to wait for the 1.0.0 release to further implement multiplayer.

**Comment by mbrlabs** [comment]
Wow, this is really impressive!
I never messed with multiplayer in Godot. Is this a peer-to-peer connection? Does it require you to be in the same network?

**Reply by Gamespleasure** [reply]
I am using a server-to-peer model for multiplayer, which means that it's possible to export the application as a dedicated server, the host can either participate with other users or just host the session if they have open ports on their router.

**Reply by mbrlabs** [reply]
Yeah, that's what i thought. Still very cool, good to know that it's doable. 👍

It's probably best to wait until we hit a somwhat stable release (like 1.0) until we add multiplayer. I'm definitely not against adding it...seems very useful!

**Reply by MrApplejuice** [reply]
Super-duper useful. I was also thinking of that. Would make it super-easy to use a tablet computer, for example, and then quickly walk to your desktop to take your notes and do something else with this. Collaboration is the other obvious usecase, I think.
